### PR TITLE
pre-commit: Add hook to add sourceheaders automatically

### PR DESCRIPTION
**Pull request by Holzhaus**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -50,3 +50,7 @@ repos:
       -D,
       warnings,
     ]
+- repo: https://github.com/Holzhaus/sourceheaders
+  rev: 41391140ca6ac7c3dfddae090353d183cbeadb48
+  hooks:
+  - id: sourceheaders
```
Comment (Swiftb0y) [inline .pre-commit-config.yaml:56]: would've wished this was written in Rust (cuz u know, [oxidize it](https://wiki.mozilla.org/Oxidation)) but python makes sense too for pre-commit. ;)

File: build.rs [hunk 2]
```diff
@@ -1,3 +1,10 @@
+// Copyright (c) 2022 Jan Holthuis <jan.holthuis@rub.de>
+//
+// This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0. If a copy
+// of the MPL was not distributed with this file, You can obtain one at
+// http://mozilla.org/MPL/2.0/.
+//
+// SPDX-License-Identifier: MPL-2.0
 use glob::glob;
```
Comment (Swiftb0y) [inline build.rs:8]: I think it would be a nice addition to `sourceheaders` if it put at least one newline after the license header.

Comment (Swiftb0y) [reply build.rs:8]: Or you can just do that yourself I guess since its purely a style choice.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (Holzhaus) [reply .pre-commit-config.yaml:56]: I thought about it, but unfortunately rust code requires a system install of rust on the host, so I decided against it.

File: build.rs (hunk 2 above)
Comment (Holzhaus) [reply build.rs]: Done.

**Review by Swiftb0y — APPROVED** [review]

**Pull request merged**